src/relay/pass/util.cc

```diff
@@ -0,0 +1,118 @@
+/*!
+ * Copyright (c) 2018 by Contributors
+ *
+ * \file util.cc
+ *
+ * \brief simple util for relay.
+ */
+#include <tvm/relay/pass.h>
+#include <tvm/relay/expr_functor.h>
+#include "./type_visitor.h"
+
+namespace tvm {
+namespace relay {
+
+class FreeVar;
+class FreeTypeVar : private TypeVisitor<> {
+  std::unordered_set<TypeParam, NodeHash, NodeEqual> * free;
+  std::unordered_set<TypeParam, NodeHash, NodeEqual> * bound;
+  FreeTypeVar(std::unordered_set<TypeParam, NodeHash, NodeEqual> * free,
```

Comment:
style std::unordered_set<TypeParam, NodeHash, NodeEqual>* free_vars(free is a bad name as it reads like the malloc/free